server/src/main/java/org/apache/druid/server/coordinator/config/CoordinatorKillConfigs.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.coordinator.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.common.config.Configs;
+import org.joda.time.Duration;
+
+public class CoordinatorKillConfigs
+{
+  public static CoordinatorKillConfigs STANDARD
+      = new CoordinatorKillConfigs(null, null, null, null, null, null, null, null, null, null, null, null);
+
+  @JsonProperty("supervisor")
+  private final MetadataCleanupConfig supervisors;
+
+  @JsonProperty("audit")
+  private final MetadataCleanupConfig audit;
+
+  @JsonProperty("datasource")
+  private final MetadataCleanupConfig datasource;
+
+  @JsonProperty("rule")
+  private final MetadataCleanupConfig rules;
+
+  @JsonProperty("compaction")
+  private final MetadataCleanupConfig compaction;
+
+  @JsonProperty("pendingSegments")
+  private final MetadataCleanupConfig pendingSegments;
+
+  // Raw configs for killing unused segments
+  // These have been added as fields because KillUnusedSegmentsConfig is initialized lazily
+  @JsonProperty("on")
+  private final Boolean killUnusedEnabled;
+
+  @JsonProperty("period")
+  private final Duration killUnusedPeriod;
+
+  @JsonProperty("durationToRetain")
+  private final Duration killUnusedDurationToRetain;
+
+  @JsonProperty("ignoreDurationToRetain")
+  private final Boolean killUnusedIgnoreDurationToRetain;
+
+  @JsonProperty("bufferPeriod")
+  private final Duration killUnusedBufferPeriod;
+
+  @JsonProperty("maxSegments")
+  private final Integer killUnusedMaxSegments;
+
+  @JsonCreator
+  public CoordinatorKillConfigs(
+      @JsonProperty("pendingSegments") MetadataCleanupConfig pendingSegments,
+      @JsonProperty("supervisor") MetadataCleanupConfig supervisors,
+      @JsonProperty("audit") MetadataCleanupConfig audit,
+      @JsonProperty("datasource") MetadataCleanupConfig datasource,
+      @JsonProperty("rule") MetadataCleanupConfig rules,
+      @JsonProperty("compaction") MetadataCleanupConfig compaction,
+      // Configs for cleanup of unused segments
```

Comment:
:( wish he had a namespace prefix for unused segments similar to the others...

Reply:
Yeah, I wanted to use a namespace prefix for unused segments and deprecate these older config paths. But that would result in unnecessary work for an operator down the line when the deprecated config paths are completely removed.